editor/store/selectors.js

```diff
@@ -428,8 +428,8 @@ export const getBlock = createSelector(
 			innerBlocks: getBlocks( state, uid ),
 		};
 	},
-	( state, uid ) => [
-		get( state, [ 'editor', 'present', 'blocksByUid', uid ] ),
```

Comment:
Don't you think the fix should be to add a dependency to `blockOrder[uid]` instead?